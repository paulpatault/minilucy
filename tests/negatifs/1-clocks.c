#include <stdlib.h>

#include <printf.h>

#include <unistd.h>

int int_read() {
  int var;
  scanf("%d", &var);
  return var;
}

enum inductive_bool {
  FALSE,
  TRUE
};

struct f_mem {
  int aux__2_next2;
  int aux__4_next1;
};

void f_init (struct f_mem* mem) {
  mem->aux__2_next2 = 1;
  
  mem->aux__4_next1 = 0;
}

int f (struct f_mem* mem, int i, int c) {
  int aux__5;
  int aux__4;
  int aux__3;
  int aux__2;
  int aux__1;
  int t;
  int x;
  int call_1;
  
  x = i;
  
  aux__1 = 0;
  
  aux__2 = mem->aux__2_next2;
  
  aux__4 = mem->aux__4_next1;
  
  t = aux__2 ? 0 : aux__4;
  
  aux__3 = t;
  
  printf("%d", t);
  
  aux__5 = call_1;
  
  mem->aux__2_next2 = aux__1;
  
  mem->aux__4_next1 = aux__3;
  
  return x;
}

int main (int argc, char* argv[]) {
  struct f_mem mem;
  int argv_0;
  int argv_1;
  int res;
  
  f_init(&(mem));
  
  while (1) {
    argv_0 = int_read();
    
    argv_1 = int_read();
    
    res = f(&(mem), argv_0, argv_1);
    
    printf("%d", res);
    
    fflush(0);
    
    usleep(333333);
  };
}
