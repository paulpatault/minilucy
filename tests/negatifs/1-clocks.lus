node main0 (i:int; c:bool) returns (x:int);
var t: int;
let
  t = 0 fby (t when True(c));
  print(t);
  x = i;
tel
