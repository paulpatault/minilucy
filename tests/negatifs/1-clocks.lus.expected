
/**************************************/
/* Sugar automaton print              */
/**************************************/
type inductive_bool =
  | False
  | True

node main0(i : int; c : bool) returns (x : int);
var print_var__1 : int; t : int;
let
  (t) = (0) -> (pre (t when True(c)));
  (print_var__1) = print(""t);
  (x) = i;
tel

/**************************************/
/* Typed ast                          */
/**************************************/
type inductive_bool =
  | False
  | True

node main0__10(i__8 : int; c__9 : bool) returns (x__5 : int)
var print_var__1__6: int; t__7: int;
let
  (t__7) = ite(true fby false, 0, 0 fby t__7 when True(c__9));
  (print_var__1__6) = print("", t__7);
  (x__5) = i__8;
tel

/**************************************/
/* Normalized ast                     */
/**************************************/
type inductive_bool =
  | False
  | True

node main0__10(i__8 : int; c__9 : bool) returns (x__5 : int)
var aux'5__15: int; aux'4__14: int; aux'3__13: int; aux'2__12: bool;
    aux'1__11: bool; print_var__1__6: int; t__7: int;
let
  (aux'4__14) = 0 fby aux'3__13;
  (aux'3__13) = t__7 when True(c__9);
  (aux'2__12) = true fby aux'1__11;
  (aux'1__11) = false;
  (t__7) = ite(aux'2__12, 0, aux'4__14);
  (aux'5__15) = print("", t__7);
  (print_var__1__6) = aux'5__15;
  (x__5) = i__8;
tel

[31m[1mClocking error: [0mThe expected clock is Base, got Base on True(c__9)

